DataStructInterfaces/IMyBinaryTree.cs

```diff
@@ -2,5 +2,5 @@ namespace DataStructInterfaces;
 
 public interface IMyBinaryTree
```

Comment:
actually, you can remove this interface, and use IMyCollection in the BinaryTree, with T as a IComparable